tasks/eva_2641/write_annotations.groovy

```diff
@@ -0,0 +1,102 @@
+@Grab(group='uk.ac.ebi.eva', module='eva-pipeline', version='2.8-SNAPSHOT')
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.batch.item.ItemWriter
+import org.springframework.batch.item.support.CompositeItemWriter
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.mongodb.core.MongoOperations
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+
+import uk.ac.ebi.eva.commons.models.metadata.AnnotationMetadata
+import uk.ac.ebi.eva.commons.models.mongo.entity.Annotation
+import uk.ac.ebi.eva.pipeline.configuration.io.writers.AnnotationCompositeWriterConfiguration
+import uk.ac.ebi.eva.pipeline.io.writers.AnnotationInVariantMongoWriter
+import uk.ac.ebi.eva.pipeline.io.writers.AnnotationMongoWriter
+
+@Component
+@Import(value=[AnnotationCompositeWriterConfiguration.class])
+class MainApp implements CommandLineRunner {
+
+    @Autowired
+    private MongoOperations mongoOperations
+
+    private ItemWriter<List<Annotation>> variantAnnotationItemWriter
+
+    private ItemWriter<List<Annotation>> annotationItemWriter
+
+    private ItemWriter<List<Annotation>> annotationWriter
+
+    @Value('${app.vep.version}')
+    private String vepVersion
+
+    @Value('${app.vep.cache.version}')
+    private String vepCacheVersion
+
+    @Value('${db.collections.variants.name}')
+    private String variantCollectionName
+
+    @Value('${db.collections.annotation-metadata.name}')
+    private String metadataCollectionName
+
+    @Value('${db.collections.annotations.name}')
+    private String annotationCollectionName
+
+    private final String tempCollectionName = "eva2641_temp_annotations"
+
+    private static Logger logger = LoggerFactory.getLogger(MainApp.class)
+
+    void run(String... args) {
+        logger.info("VEP version {}, cache version {}", vepVersion, vepCacheVersion)
+
+        annotationItemWriter = annotationItemWriter(mongoOperations)
+        variantAnnotationItemWriter = variantAnnotationItemWriter(mongoOperations)
+        annotationWriter = compositeAnnotationItemWriter()
+
+        List<Annotation> annotations = readAnnotationsFromTempCollection()
+        logger.info("Read {} annotations", annotations.size())
+
+        annotationWriter.write(Collections.singletonList(annotations))
```

Comment:
singletonList can hold only one element. Did you mean to put in unModifiableList here?

Reply:
It should be a one-element list containing a list of annotations...  In the pipeline I think the inner list would be a chunk of 200 annotations that are themselves written in batches, I wasn't sure if that was necessary in this script, as we're not actually passing variants through VEP or anything.

Reply:
Got it... I see your point but let's keep the method signature as it is even if it means using a singletonList here...